app/src/main/java/ca/marshallasch/veil/MainActivity.java

```diff
@@ -354,4 +226,24 @@ public void setCurrentUser(@NonNull DhtProto.User currentUser)
     {
         this.currentUser = currentUser;
     }
+
+
+    /**
+     * Sends messages to {@link VeilService} to do work with.
+     * @param view optional view for setting objects in view
+     * @param command non optional command int defined by static strings in {@link VeilService}
+     */
+    public void sendServiceMessage(@Nullable View view, @NonNull int command){
```

Comment:
I thought it would be better to make the MainActivity the point of access of all the fragments to the service. I also can't think of an instance where you would actually attach more to the command then what is here. I am open for suggestions on how to handle this though!

Reply:
move all the message creation into the caller, What happens for the message that is in branch number #60 where it has to attach the post / comment objects?